src/main/kotlin/org/jetbrains/research/tasktracker/config/util/ConfigUtil.kt

```diff
@@ -0,0 +1,21 @@
+package org.jetbrains.research.tasktracker.config.util
+
+import com.intellij.openapi.diagnostic.Logger
+import org.jetbrains.research.tasktracker.config.BaseConfig
+import java.io.File
+
+object ConfigUtil {
+    inline fun <reified T : BaseConfig> buildConfig(
+        config: T?,
+        configName: String,
+        configFile: File,
+        configBuilder: (File) -> T
+    ): T {
+        require(config == null) {
+            "The $configName config was already parsed"
```

Comment:
I think it would be better to move config name to the BaseConfig class and just reuse it here: `config. configName`, use "" as the default value in the BaseConfig class